apps/keycloak.py

```diff
@@ -0,0 +1,379 @@
+                                "ldap.attribute": ["modifyTimestamp"],
+                                "is.mandatory.in.ldap": ["false"],
+                                "read.only": ["true"],
+                                "always.read.value.from.ldap": ["true"],
+                                "user.model.attribute": ["modifyTimestamp"],
+                            },
+                        },
+                        {
+                            "name": "group",
+                            "providerId": "group-ldap-mapper",
+                            "subComponents": {},
+                            "config": {
+                                "mode": ["READ_ONLY"],
+                                "membership.attribute.type": ["UID"],
+                                "user.roles.retrieve.strategy": [
+                                    "LOAD_GROUPS_BY_MEMBER_ATTRIBUTE"
+                                ],
+                                "group.name.ldap.attribute": ["cn"],
+                                "preserve.group.inheritance": ["false"],
+                                "ignore.missing.groups": ["false"],
+                                "membership.user.ldap.attribute": ["uid"],
+                                "membership.ldap.attribute": ["memberUid"],
+                                "group.object.classes": ["posixGroup"],
+                                "groups.dn": ["ou=Group,dc=ocf,dc=Berkeley,dc=EDU"],
+                                "memberof.ldap.attribute": ["memberOf"],
+                                "drop.non.existing.groups.during.sync": ["false"],
+                            },
+                        },
+                    ]
+                },
+            }
+        ],
+    },
+}
+
+
+krb5_conf = """
+[libdefaults]
+  default_realm = OCF.BERKELEY.EDU
+
+# The following krb5.conf variables are only for MIT Kerberos.
+  krb4_config = /etc/krb.conf
+  krb4_realms = /etc/krb.realms
+  kdc_timesync = 1
+  ccache_type = 4
+  forwardable = true
+  proxiable = true
+
+# The following libdefaults parameters are only for Heimdal Kerberos.
+  v4_instance_resolve = false
+  v4_name_convert = {
+    host = {
+      rcmd = host
+      ftp = ftp
+    }
+    plain = {
+      something = something-else
+    }
+  }
+  fcc-mit-ticketflags = true
+
+[realms]
+  OCF.BERKELEY.EDU = {
+    kdc = kerberos.ocf.berkeley.edu
+    admin_server = kerberos.ocf.berkeley.edu
+  }
+
+[domain_realm]
+  .ocf.berkeley.edu = OCF.BERKELEY.EDU
+  ocf.berkeley.edu = OCF.BERKELEY.EDU
+
+[login]
+  krb4_convert = true
+  krb4_get_tickets = false
+"""
+
+helm_values = {
+    "auth": {"existingSecret": "keycloak", "existingSecretKey": "admin-password"},
+    "production": True,
+    "proxy": "edge",
+    "httpRelativePath": "/",
+    "replicaCount": 2,
+    "ingress": {
+        "enabled": True,
+        "ingressClassName": "contour",
+        "annotations": {
+            "cert-manager.io/cluster-issuer": "letsencrypt",
```

Comment:
`config.defaults.certManagerIssuer`

Reply:
(we really should have a way to get all of the required annotations in one go, but that can be a transpire PR later)